File: Musixmatch_lyrics_API_App/app/src/main/java/alexejantonov/com/musixmatch_lyrics_api_app/ui/track_details/TrackDetailsActivity.java
package alexejantonov.com.musixmatch_lyrics_api_app.ui.track_details;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.RequestManager;

import alexejantonov.com.musixmatch_lyrics_api_app.R;
import alexejantonov.com.musixmatch_lyrics_api_app.api.entities.track.Track;

import static alexejantonov.com.musixmatch_lyrics_api_app.ui.track_details.TrackDetailsScreenContract.View;

public class TrackDetailsActivity extends AppCompatActivity implements View {

	private static final String EXTRA_TRACK = TrackDetailsActivity.class.getSimpleName() + "track";

	private Track track;
	private RequestManager imageRequestManager;
	private TrackDetailsPresenter presenter = new TrackDetailsPresenter();

	private TextView trackName;
	private TextView trackAlbum;
	private ImageView albumCover;
	private TextView lyrics;
	private Toolbar toolbar;

	public static Intent newIntent(Context packageContext, Track track) {
		Intent i = new Intent(packageContext, TrackDetailsActivity.class);
		i.putExtra(EXTRA_TRACK, track);
		return i;
	}

	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_track_details);

		track = (Track) getIntent().getSerializableExtra(EXTRA_TRACK);

		toolbar = findViewById(R.id.toolbar);
		setSupportActionBar(toolbar);
		getSupportActionBar().setTitle(getString(R.string.lyrics_of) + "\"" + track.getTrackName() + "\"");
		toolbar.setNavigationIcon(R.drawable.ic_arrow_back_white_24dp);
		toolbar.setNavigationOnClickListener(v -> onBackPressed());

		trackName = findViewById(R.id.trackName);
		trackAlbum = findViewById(R.id.trackAlbum);
		albumCover = findViewById(R.id.albumCover);
		lyrics = findViewById(R.id.lyrics);

		trackName.setText("\"" + track.getTrackName() + "\"");
		trackAlbum.setText(getString(R.string.album) + " \"" + track.getAlbumName() + "\"");
		imageRequestManager = Glide.with(this);
		imageRequestManager.load(track.getAlbumCover()).into(albumCover);

		presenter.setTrackId(String.valueOf(track.getTrackId()));
		presenter.onAttach(this);
	}

	@Override
	protected void onDestroy() {
		super.onDestroy();
		presenter.onDetach();
	}

	@Override
	public void showData(String lyricsText) {
		lyrics.setText(lyricsText);
	}
}

File: Musixmatch_lyrics_API_App/app/src/main/java/alexejantonov/com/musixmatch_lyrics_api_app/ui/track_details/TrackDetailsActivity.kt
package alexejantonov.com.musixmatch_lyrics_api_app.ui.track_details

import alexejantonov.com.musixmatch_lyrics_api_app.R
import alexejantonov.com.musixmatch_lyrics_api_app.api.entities.track.Track
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import com.arellomobile.mvp.MvpAppCompatActivity
import com.arellomobile.mvp.presenter.InjectPresenter
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.activity_track_details.albumCover
import kotlinx.android.synthetic.main.activity_track_details.lyrics
import kotlinx.android.synthetic.main.activity_track_details.progressBar
import kotlinx.android.synthetic.main.activity_track_details.toolbar
import kotlinx.android.synthetic.main.activity_track_details.trackAlbum
import kotlinx.android.synthetic.main.activity_track_details.trackName

class TrackDetailsActivity : MvpAppCompatActivity(), TrackDetailsView {
  companion object {
    private val EXTRA_TRACK = TrackDetailsActivity::class.java.simpleName + "track"

    fun newIntent(context: Context, track: Track) = Intent(context, TrackDetailsActivity::class.java).apply {
      putExtra(EXTRA_TRACK, track)
    }
  }

  private var track: Track? = null

  @InjectPresenter
  lateinit var presenter: TrackDetailsPresenter

  public override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_track_details)

    track = intent.getSerializableExtra(EXTRA_TRACK) as Track

    setSupportActionBar(toolbar)
    supportActionBar?.title = getString(R.string.lyrics_of) + "\"" + track?.trackName + "\""
    toolbar.apply {
      setNavigationIcon(R.drawable.ic_arrow_back_white_24dp)
      setNavigationOnClickListener { onBackPressed() }
    }

    trackName.text = track?.trackName
    trackAlbum.text = String.format(getString(R.string.album), track?.albumName)
    Glide.with(this).load(track?.albumCover).into(albumCover)

    presenter.setTrackId(track!!.trackId.toString())
  }

  override fun showData(lyricsText: String) {
    lyrics.text = lyricsText
  }

  override fun showLoading() {
    progressBar.visibility = View.VISIBLE
  }

  override fun hideLoading() {
    progressBar.visibility = View.GONE
  }
}
